### ISSUE TITLE
What's up with all the ^H characters (was opam --help)

### ISSUE THREAD

**Issue by dbuenzli** [dbuenzli/cmdliner#126]
@cpitclaudel following up with https://github.com/ocaml/opam/issues/4812.

So your system should invoke `groff`. For some reason it doesn't doesn't seem to work.

Since you are on linux I suspect there is no problem in the groff output (e.g. newline translation business). 

So could you try maybe try variations of the `groff` invocation to see if you get a clean result (I have no recall of why that `-P-c` is for here 😬). E.g. 

```
opam --help=groff | groff -Tascii
opam --help=groff | groff -Tascii -man
opam --help=groff | groff -Tutf8 

```

Also could you make sure that `groff -Tascii -man $(man --path grep) | less` doesn't exhibit the same behaviour.

**Comment by dbuenzli** [comment]
A but wait I missed an important bit: `When I run opam help in Emacs I get a lot of ^H characters"`. 

Before I get crazy looking into groff could you first check there's no problem in your terminal :-) If there's no problem there I'll close this…

There's a lot different quirks depending on how and where you run commands in emacs and I won't debug that. 

If you are running that in compilation mode you may try something like: 

```
(require 'xterm-color)
(defun my/advice-compilation-filter (f proc string)
  (funcall f proc (xterm-color-filter string)))
(advice-add 'compilation-filter :around #'my/advice-compilation-filter)
```

**Comment by cpitclaudel** [comment]
> Before I get crazy looking into groff could you first check there's no problem in your terminal :-)

That's why I posted the terminal output in the original issue: in a sense, the output is all right: these `^H` characters are backspaces, so what it's doing is printing two characters, then deleting one, and so on. So the final output is valid (and it would look fine if Emacs interpreted those backspaces) — it's just weird.

>

On my system it needs to be `zcat $(man --path grep) | groff -Tascii -man | less` (the man pages are compressed).  And it has the exact same problem in Emacs as OPAM's output.  The unexpected `^H` characters can be counted like this, btw:

```
$ zcat $(man --path grep) | groff -Tascii -man | grep -o $'\b' | wc -l
3264
```

Bottom line: it's not a cmdliner issue, just a weird thing that groff does.  Hence we can probably close this… sorry for the noise!

**Comment by dbuenzli** [comment]
> Hence we can probably close this… sorry for the noise!

No worries. 

If you find a way to avoid this, be my guest. Personally I'm not in a mood to go hunt what `groff` may be doing :-)
